pkg/database/billing.go

```diff
@@ -29,48 +36,61 @@ type Billing interface {
 	collc := cosmosdb.NewCollectionClient(dbc, dbName)
 
-	triggers := []*cosmosdb.Trigger{
+	triggerResources := []*armcosmos.SQLTriggerResource{
 		{
-			ID:               "setCreationBillingTimeStamp",
-			TriggerOperation: cosmosdb.TriggerOperationCreate,
-			TriggerType:      cosmosdb.TriggerTypePre,
-			Body: `function trigger() {
-	var request = getContext().getRequest();
-	var body = request.getBody();
-	var date = new Date();
-	var now = Math.floor(date.getTime() / 1000);
-	var billingBody = body["billing"];
-	if (!billingBody["creationTime"]) {
-		billingBody["creationTime"] = now;
-	}
-	request.setBody(body);
-}`,
+			ID: to.Ptr("setCreationBillingTimeStamp"),
+			Body: to.Ptr(`function trigger() {
+				var request = getContext().getRequest();
+				var body = request.getBody();
+				var date = new Date();
+				var now = Math.floor(date.getTime() / 1000);
+				var billingBody = body["billing"];
+				if (!billingBody["creationTime"]) {
+					billingBody["creationTime"] = now;
+				}
+				request.setBody(body);
+			}`),
+			TriggerOperation: to.Ptr(armcosmos.TriggerOperation("Create")),
+			TriggerType:      to.Ptr(armcosmos.TriggerType("Pre")),
 		},
 		{
-			ID:               "setDeletionBillingTimeStamp",
-			TriggerOperation: cosmosdb.TriggerOperationReplace,
-			TriggerType:      cosmosdb.TriggerTypePre,
-			Body: `function trigger() {
-	var request = getContext().getRequest();
-	var body = request.getBody();
-	var date = new Date();
-	var now = Math.floor(date.getTime() / 1000);
-	var billingBody = body["billing"];
-	if (!billingBody["deletionTime"]) {
-		billingBody["deletionTime"] = now;
-	}
-	request.setBody(body);
-}`,
+			ID: to.Ptr("setDeletionBillingTimeStamp"),
+			Body: to.Ptr(`function trigger() {
+				var request = getContext().getRequest();
+				var body = request.getBody();
+				var date = new Date();
+				var now = Math.floor(date.getTime() / 1000);
+				var billingBody = body["billing"];
+				if (!billingBody["creationTime"]) {
+					billingBody["creationTime"] = now;
+				}
+				request.setBody(body);
+			}`),
+			TriggerOperation: to.Ptr(armcosmos.TriggerOperation("Replace")),
+			TriggerType:      to.Ptr(armcosmos.TriggerType("Pre")),
 		},
 	}
 
-	triggerc := cosmosdb.NewTriggerClient(collc, collBilling)
-	for _, trigger := range triggers {
-		_, err := triggerc.Create(ctx, trigger)
-		if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusConflict) {
+	for _, triggerResource := range triggerResources {
+		createUpdateSQLTriggerParameters := armcosmos.SQLTriggerCreateUpdateParameters{
+			Properties: &armcosmos.SQLTriggerCreateUpdateProperties{
+				Options:  &armcosmos.CreateUpdateOptions{},
+				Resource: triggerResource,
+			},
+			Location: &location,
+		}
+		ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+		defer cancel()
```

Comment:
Do we have an over-arching context assocaited with this function already?  is the old method `triggerc.Create(ctx, trigger)` blocking vs now we have to poll?